From f2caeaab4558975e38b05ec41b9470c148c7d0da Mon Sep 17 00:00:00 2001
From: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Date: Mon, 17 Mar 2025 11:36:30 +0100
Subject: [PATCH] arm64: dts: st: lptimer: adopt fallback compatibles on
 stm32mp211

Align with upstream recommendation to adopt fallback/backward compatible
strings on stm32mp211, as done for stm32mp251 [1].

[1] https://lore.kernel.org/linux-arm-kernel/20250224180150.3689638-1-fabrice.gasnier@foss.st.com/

Signed-off-by: Fabrice Gasnier <fabrice.gasnier@foss.st.com>
Change-Id: Ic1078c908589bb3712f8aee2d9d8ca4508abed2f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/445390
Reviewed-by: Olivier MOYSAN <olivier.moysan@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi | 50 +++++++++++++-------------
 1 file changed, 25 insertions(+), 25 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp211.dtsi b/arch/arm64/boot/dts/st/stm32mp211.dtsi
index 3e1cffe73ebc..a142195a712a 100644
--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -669,7 +669,7 @@ timer@13 {
 			};
 
 			lptimer1: timer@40090000 {
-				compatible = "st,stm32mp21-lptimer";
+				compatible = "st,stm32mp21-lptimer", "st,stm32-lptimer";
 				reg = <0x40090000 0x0 0x400>;
 				interrupts-extended = <&exti1 47 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM1>;
@@ -682,30 +682,30 @@ lptimer1: timer@40090000 {
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp21-lptimer-counter";
+					compatible = "st,stm32mp21-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp21-pwm-lp";
+					compatible = "st,stm32mp21-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp21-lptimer-timer";
+					compatible = "st,stm32mp21-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@0 {
-					compatible = "st,stm32mp21-lptimer-trigger";
+					compatible = "st,stm32mp21-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <0>;
 					status = "disabled";
 				};
 			};
 
 			lptimer2: timer@400a0000 {
-				compatible = "st,stm32mp21-lptimer";
+				compatible = "st,stm32mp21-lptimer", "st,stm32-lptimer";
 				reg = <0x400a0000 0x0 0x400>;
 				interrupts-extended = <&exti1 48 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM2>;
@@ -718,23 +718,23 @@ lptimer2: timer@400a0000 {
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp21-lptimer-counter";
+					compatible = "st,stm32mp21-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp21-pwm-lp";
+					compatible = "st,stm32mp21-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp21-lptimer-timer";
+					compatible = "st,stm32mp21-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@1 {
-					compatible = "st,stm32mp21-lptimer-trigger";
+					compatible = "st,stm32mp21-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <1>;
 					status = "disabled";
 				};
@@ -1699,7 +1699,7 @@ i3c3: i3c@46080000 {
 			};
 
 			lptimer3: timer@46050000 {
-				compatible = "st,stm32mp21-lptimer";
+				compatible = "st,stm32mp21-lptimer", "st,stm32-lptimer";
 				reg = <0x46050000 0x0 0x400>;
 				interrupts-extended = <&exti2 29 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM3>;
@@ -1711,30 +1711,30 @@ lptimer3: timer@46050000 {
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp21-lptimer-counter";
+					compatible = "st,stm32mp21-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp21-pwm-lp";
+					compatible = "st,stm32mp21-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp21-lptimer-timer";
+					compatible = "st,stm32mp21-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@2 {
-					compatible = "st,stm32mp21-lptimer-trigger";
+					compatible = "st,stm32mp21-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <2>;
 					status = "disabled";
 				};
 			};
 
 			lptimer4: timer@46060000 {
-				compatible = "st,stm32mp21-lptimer";
+				compatible = "st,stm32mp21-lptimer", "st,stm32-lptimer";
 				reg = <0x46060000 0x0 0x400>;
 				interrupts-extended = <&exti2 30 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM4>;
@@ -1746,30 +1746,30 @@ lptimer4: timer@46060000 {
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp21-lptimer-counter";
+					compatible = "st,stm32mp21-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp21-pwm-lp";
+					compatible = "st,stm32mp21-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp21-lptimer-timer";
+					compatible = "st,stm32mp21-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@3 {
-					compatible = "st,stm32mp21-lptimer-trigger";
+					compatible = "st,stm32mp21-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <3>;
 					status = "disabled";
 				};
 			};
 
 			lptimer5: timer@46070000 {
-				compatible = "st,stm32mp21-lptimer";
+				compatible = "st,stm32mp21-lptimer", "st,stm32-lptimer";
 				reg = <0x46070000 0x0 0x400>;
 				interrupts-extended = <&exti2 31 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_KER_LPTIM5>;
@@ -1781,23 +1781,23 @@ lptimer5: timer@46070000 {
 				status = "disabled";
 
 				counter {
-					compatible = "st,stm32mp21-lptimer-counter";
+					compatible = "st,stm32mp21-lptimer-counter", "st,stm32-lptimer-counter";
 					status = "disabled";
 				};
 
 				pwm {
-					compatible = "st,stm32mp21-pwm-lp";
+					compatible = "st,stm32mp21-pwm-lp", "st,stm32-pwm-lp";
 					#pwm-cells = <3>;
 					status = "disabled";
 				};
 
 				timer {
-					compatible = "st,stm32mp21-lptimer-timer";
+					compatible = "st,stm32mp21-lptimer-timer", "st,stm32-lptimer-timer";
 					status = "disabled";
 				};
 
 				trigger@4 {
-					compatible = "st,stm32mp21-lptimer-trigger";
+					compatible = "st,stm32mp21-lptimer-trigger", "st,stm32-lptimer-trigger";
 					reg = <4>;
 					status = "disabled";
 				};
-- 
2.50.1

